changes/2020-07-14_multi-keyring-require-generation/change.md

```diff
@@ -0,0 +1,114 @@
+
+| Feature                                                                                                                                               |
+| ----------------------------------------------------------------------------------------------------------------------------------------------------- |
+| [Multi-Keyring](https://github.com/awslabs/aws-encryption-sdk-specification/blob/31b0534c4259aad365f048b73231545583389c67/framework/multi-keyring.md) |
+| [Multi-Keyring Generator](https://github.com/awslabs/aws-encryption-sdk-specification/issues/114)                                                     |
+
+## Affected Specifications
+
+This serves as a reference of all specification documents that this change affects.
+
+| Specification                                     |
+| ------------------------------------------------- |
+| [Multi-Keyring](../../framework/multi-keyring.md) |
+
+## Affected Implementations
+
+This serves as a reference for all implementations that this change affects.
+
+| Language   | Repository                                                                            |
+| ---------- | ------------------------------------------------------------------------------------- |
+| C          | [aws-encryption-sdk-c](https://github.com/aws/aws-encryption-sdk-c)                   |
+| Javascript | [aws-encryption-sdk-javascript](https://github.com/aws/aws-encryption-sdk-javascript) |
+
+This change additionally affects the unreleased AWS KMS keyring implementations for Java and Python.
+
+## Definitions
+
+### Conventions used in this document
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Summary
+
+The multi-keyring has a clearly configured intent of which keyring SHOULD be used to generate the data key,
+but this is not actually enforced during the encryption operation.
+This change strengthens the requirements on multi-keyrings such that if a generator is configured,
+it MUST generate the data key.
+
+## Out of Scope
+
+If a multi-keyring is configured with another multi-keyring as its generator,
+and this second multi-keyring has no generator,
+the outer multi-keyring cannot ever successfully complete an OnEncrypt operation.
+This is one of several examples of keyring configurations
+that cannot satisfy the requirements of an operation.
+Detecting at keyring configuration time that encryption/decryption will always fail
+because keyrings are not capable of meeting their requirements
+is [tracked separately](https://github.com/awslabs/aws-encryption-sdk-specification/issues/144).
+
+## Motivation
+
+If we do not make a guarantee that the generator will generate,
+then its usefulness is limited.
+Customers may have a compliance requirement that all data keys MUST be generated by a specific HSM,
```

Comment:
wording: Avoid "may" when it is not "MAY". In this case I would use "might", "could", or similar.